File: packages/ts/src/modules/regions.ts
import type { HubflyClient } from '../client.js';
import type { ApiResponse, Region } from '../types.js';

export class RegionsModule {
  constructor(private readonly client: HubflyClient) {}

  /**
   * List all available regions
   */
  public async list(): Promise<ApiResponse<Region[]>> {
    return this.client.request<Region[]>('/api/v1/regions');
  }
}

File: packages/go/regions.go
package hubfly

import "context"

type RegionsService struct {
	client *Client
}

// List returns all available regions
func (s *RegionsService) List(ctx context.Context) (*Response[[]Region], error) {
	return Request[[]Region](ctx, s.client, "GET", "/api/v1/regions", nil)
}
